tests/maker.rs

```diff
@@ -330,3 +395,152 @@ fn test_maker_cli(maker_cli: &MakerCli, rx: &Receiver<String>) {
     await_message(rx, "Maker is shutting down");
     await_message(rx, "Maker Server is shut down successfully");
 }
+
+fn test_bitcoin_backend_connection(maker_cli: &mut MakerCli) {
+    println!("TEST STARTING: Bitcoin Backend Connection");
+    let (rx, mut maker) = maker_cli.start_makerd();
+
+    await_message(&rx, "Server Setup completed!!");
+    println!("Maker started with connection to Bitcoin backend");
+
+    await_message_timeout(&rx, "Swap Liquidity:", Duration::from_secs(60));
+    println!("✅ Verified Bitcoin connection with successful liquidity check");
+
+    // Stop bitcoind
+    println!("Stopping bitcoind to test disconnection handling...");
+    maker_cli.bitcoind.stop().unwrap();
+
+    await_message_timeout(&rx, "RPC Connection failed", Duration::from_secs(20));
+    println!("✅ Verified maker detects Bitcoin backend disconnection");
+
+    // cleanup an new bitcoind instance
+    maker.kill().unwrap();
+    maker.wait().unwrap();
```

Comment:
Don't kill the maker here. Just restart the bitcoind. And this should log the message: "Bitcoin Core RPC connection is live.". 

Assert that log.